### DISCUSSION TITLE
Can you run the Docker container without checking out the source repo?

### DISCUSSION THREAD

**Discussion by TheFriendlyCoder** [netbox-community/netbox-docker#628] [General]
Good day all.

We are using this container to set up a test environment for Netbox to do ad-hoc and automated testing with, and we would like to be able to run the container directly from Dockerhub. However, based on the docs associated with the project it seems that this is not possible. We first have to check out the associated Github repository first and use files / content from the source repo to run the container. Shouldn't there be a way to run the container directly?

If I've missed something in the docs and there is actually a way to do this, could someone please let me know how?

**Comment by cimnine** [comment]
Sure, you can do that. For instance, there are several people and organizations that run NetBox Docker in Kubernetes.

Just make sure that you define all the required configurations (either via env variables or file mounts), possibly (but for a test environment not necessarily) the volumes, definitely the dependencies (Redis and Postgres containers, possibly worker container).
You find all the information to replicate the environment in the `docker-compose.yml` file (and the files it references, such as `netbox.env`).

Checking out the repository is just the quickest and easiest way to start – certainly for the folks with little to no Docker experience.

**Reply by TheFriendlyCoder** [reply]
Thanks for the prompt reply. Do you have any links to docs or examples on how to set up a fully running Netbox docker container without using the source repo? When I looked over the docker-compose.yml file found in the source repo I noticed there were several folders being mounted as volumes to the container and that some of those folders contained dozens of files. Trying to figure out which ones may be necessary for a simple environment setup and which are not seems to be non trivial. 

Any information you can provide would be greatly appreciated.

**Reply by kkthxbye-code** [reply]
I'm pretty sure only configuration.py needs to be mapped to /etc/netbox/config/configuration.py to run the image. Startup scripts and initializers er both netbox-docker specific optional features, they can be disabled via. SKIP_STARTUP_SCRIPTS. Scripts and reports are optional netbox features. And you don't have to map media files to the host/presistant volume if you don't care about saving them.

The rest is configured via. environment variables.

**Reply by cimnine** [reply]
All folders are just mounted for convenience. All relevant files are present in the container image as well. Check the `Dockerfile` for reference. (We mount the files in the compose file because that makes adding custom configuration and editing existing files easier.)

**Reply by TheFriendlyCoder** [reply]
Thanks for the info. We'll give this a test in the next couple of days and I'll let you know how we make out.
